allocation.py

```diff
@@ -22,6 +27,13 @@ def get_rec_name(self, name):
     def search_rec_name(cls, name, clause):
         return [('employee.rec_name',) + tuple(clause[1:])]
 
+    def get_company(self, name):
+        return self.work.company.id if self.work.company else None
```

Comment:
As it will be come an on_change_with it will need to check that both "self.work" and "self.work.company" have a value.